### PR TITLE
Assigment 4

### PR DESCRIPTION
**Pull request by JavierSabin**

### CODE REVIEW

File: Assignment4/JavierSabin-190299/task06.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1hSv61ejXESrp34CUz1PqZCsoA43tzc8y
+
+**Task 06: Modifying RDF(s)**
+"""
+
+!pip install rdflib
```
Comment (dgarijo) [inline Assignment4/JavierSabin-190299/task06.py]: won't work in python. This is notebook magic

File: Assignment4/JavierSabin-190299/task06.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+VCARD = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+g.add((ns.JaneSmith, VCARD.FN, fullName))
+g.add((ns.JaneSmith, VCARD.Given, given))
+g.add((ns.JaneSmith, VCARD.Family, family))
+for s, p, o in g:
+  print(s,p,o)
+
+"""TASK 6.5: Add UPM as the university where John Smith works"""
+g.add((ns.UPM, RDF.type, ns.University))
+g.add((ns.JaneSmith, VCARD.Work, ns.UPM))
```
Comment (dgarijo) [inline Assignment4/JavierSabin-190299/task06.py]: Work does not exist as a property in vcard. Create your own in your own ns

File: Assignment4/JavierSabin-190299/task07.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/resources/example6.rdf", format="xml")
+
+"""TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL"""
+# RDF
+ns = Namespace("http://somewhere#")
+for s,p,o in g.triples((None, RDFS.subClassOf, ns.Person)):
+    print(s)
+for s2,p2,o2 in g.triples((None, RDFS.subClassOf, s)):
+    print(s2)
```
Comment (dgarijo) [inline Assignment4/JavierSabin-190299/task07.py]: this is not a well formed loop for what you are intending.
For the purpose of the exercise, you only need the first one

File: Assignment4/JavierSabin-190299/task07.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  for s2, p2, o2 in g.triples((None, RDF.type, s)):
+    print(s2)
+
+
+# SPARQL
+query2 = prepareQuery('''
+  SELECT DISTINCT ?Person
+  WHERE{
+    {?Person rdf:type ns:Person} UNION
+	  {?Person2 (rdfs:subClassOf/rdfs:subClassOf*) ns:Person .
```
Comment (dgarijo) [inline Assignment4/JavierSabin-190299/task07.py]: why the parenthesis? not needed

File: Assignment4/JavierSabin-190299/task08.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    ?Subject ?Predicate ?Object
+    FILTER(?Predicate=vcard:FN || ?Predicate=vcard:Given || ?Predicate=vcard:EMAIL)
+  }
+  ''',
+  initNs = { "vcard": vcard}
+  )
+
+for r in g2.query(query2):
+  print(r)
+  g1.add(r)
```
Comment (dgarijo) [inline Assignment4/JavierSabin-190299/task08.py]: This is not correct. You are adding all the subjects of g2 into g1. The question is asking to complete the elements of g1 from those that have more information in g2.

**Review by dgarijo — CHANGES_REQUESTED** [review]

**Comment by JavierSabin** [comment]
Errors have been corrected

File: Assignment4/JavierSabin-190299/task08.py (hunk 5 above)
Comment (JavierSabin) [reply Assignment4/JavierSabin-190299/task08.py]: I just uploaded the corrected version

**Pull request merged**